test/rules/src/https_everywhere_checker/check_rules.py

```diff
@@ -193,16 +193,19 @@ def processUrl(self, plainUrl, task):
 			logging.info("Finished comparing %s -> %s. Rulefile: %s.",
 				plainUrl, transformedUrl, ruleFname)
 
-		return message
+		return [message, plainUrl]
 
-def disableRuleset(ruleset, problems):
+def disableRuleset(ruleset, problemsRules):
+	problems = [problem[0] for problem in problemsRules]
+	rules = [problem[1] for problem in problemsRules]
 	logging.info("Disabling ruleset %s", ruleset.filename)
 	contents = open(ruleset.filename).read()
 	# Don't bother to disable rulesets that are already disabled
 	if re.search("\bdefault_off=", contents):
 		return
-	contents = re.sub("(<ruleset [^>]*)>",
-		"\\1 default_off='failed ruleset test'>", contents)
+	for rule in rules:
+		contents = re.sub('<[ \n]*target[ \n]+host[ \n]*=[ \n]*"%s"[ \n]*\/?[ \n]*>' % (rule.split('/')[2]),
+			"", contents)
```

Comment:
Here, you'll want to remove the entire line, rather than replace just the XML tag with a blank string.